src/geom/CompoundCurve.cpp

```diff
@@ -307,5 +311,24 @@ CompoundCurve::reverseImpl() const
     return getFactory()->createCompoundCurve(std::move(reversed)).release();
 }
 
+void
+CompoundCurve::validateConstruction() const
+{
+    for (std::size_t i = 1; i < curves.size(); i++) {
+        const CoordinateXY& end = curves[i-1]->getCoordinatesRO()->back<CoordinateXY>();
+        const CoordinateXY& start = curves[i]->getCoordinatesRO()->front<CoordinateXY>();
```

Comment:
I may miss something, but it is not obvious to me that you can't insert an empty curve in a CompoundCurve, in which case back() and front() are returning something invalid. At least if using directly the CompoundCurve::CompoundCurve(std::vector<std::unique_ptr<SimpleCurve>>&& p_curves,                              const GeometryFactory& gf) constructor. That might be hard/impossible to trigger through WKT.  But maybe through WKB ?

Reply:
You're right, this is possible and would crash. Of course it couldn't be so easy.